### PR TITLE
Use globs in .projitems

### PR DESCRIPTION
**Pull request by tmat**
Makes updating and merging updates of shared projects easier.

### CODE REVIEW

File: src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.projitems [hunk 1]
```diff
@@ -10,39 +10,6 @@
...
-    <Compile Include="$(MSBuildThisFileDirectory)Helpers\EnumContextDataItem.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)Helpers\EvalResultDataItem.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)Helpers\GeneratedMetadataNames.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)Helpers\HashFunctions.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)Helpers\InternalWellKnownMemberNames.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)Helpers\MemberAndDeclarationInfo.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)Helpers\RawStringDataItem.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)Helpers\TypeHelpers.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)Helpers\ValueHelpers.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)**\*.cs" />
```
Comment (sharwell) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.projitems]: 💭 Makes me wish it used this for all the files:

```suggestion
    <Compile Include="$(MSBuildThisFileDirectory)**\*$(DefaultLanguageSourceExtension)" />
```

**Comment by tmat** [comment]
@jasonmalinowski Filed https://github.com/dotnet/project-system/issues/6010

**Comment by CyrusNajmabadi** [comment]
awesome!

**Comment by mavasani** [comment]
@tmat This is actually turning out to be painful for working with shared project due to https://github.com/dotnet/project-system/issues/6010. Moving a folder to shared project and then deleting one or more files is causing VS to remove the glob entry and replace it with explicit entries. This leads to build failure, and forces to undo the changes to projitems file, which forces the solution to be unloaded and reloaded. 

SharedProj

**Comment by mavasani** [comment]
@tmat VS is pretty much unusable due to lack of first class globing support in shared projects. Even for simple file moves between shared projects, the tooling is corrupting the projitems file and removing the glob entry, requiring undo to file changes and solution reload, and is going to be harder to track down for someone who doesn’t know about this issue.

Would you be able to restore to prior non-globing state until project system adds this support? Note you will need to revert your PR and then manually add/remove the entries from PRs that went in after your change and relied on glob entries (I at least merged couple of PRs yesterday after your merge that would need this work).

**Comment by tmat** [comment]
:(

**Pull request merged**